src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -1914,6 +1914,76 @@ public async Task PostAsync_ExpectContinue_Success(bool? expectContinue, string
             }
         }
 
+        [Fact]
+        public async Task PostAsync_TransferEncodingChunkedFalse_DoesNotSendChunked()
+        {
+            const string ExpectedContent = "Hello, expecting and continuing world.";
+            var clientCompleted = new TaskCompletionSource<bool>();
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    client.DefaultRequestHeaders.TransferEncodingChunked = false;
+
+                    var stringContent = new StringContent("test"); 
+                    stringContent.Headers.ContentLength = null;
+
+                    await client.PostAsync(uri, stringContent);
+                    clientCompleted.SetResult(true);
+                }
+            }, async server =>
+            {
+                await server.AcceptConnectionAsync(async connection =>
+                {
+                    List<string> headers = await connection.ReadRequestHeaderAsync();
+
+                    if (UseSocketsHttpHandler)
```

Comment:
Wasn't sure if I should put this in SocketsHttpHandlerTest or here with the UseSocketsHttpHandler check, can move if desired.

Reply:
Moved to there.